packages/babel-types/src/validators/isReferenced.js

```diff
@@ -7,60 +7,50 @@ export default function isReferenced(node: Object, parent: Object): boolean {
     // yes: object::NODE
     // yes: NODE::callee
     case "BindExpression":
-      return parent.object === node || parent.callee === node;
+      return true;
 
     // yes: PARENT[NODE]
     // yes: NODE.child
     // no: parent.NODE
     case "MemberExpression":
     case "JSXMemberExpression":
     case "OptionalMemberExpression":
-      if (parent.property === node && parent.computed) {
-        return true;
-      } else if (parent.object === node) {
-        return true;
-      } else {
-        return false;
+      if (parent.property === node) {
+        return !!parent.computed;
       }
+      return parent.object === node;
 
     // no: new.NODE
     // no: NODE.target
     case "MetaProperty":
       return false;
 
-    // yes: { [NODE]: "" }
-    // yes: { NODE }
-    // no: { NODE: "" }
-    case "ObjectProperty":
-      if (parent.key === node) {
-        return parent.computed;
-      }
-
     // no: let NODE = init;
     // yes: let id = NODE;
     case "VariableDeclarator":
-      return parent.id !== node;
+      return parent.init === node;
 
     // no: function NODE() {}
     // no: function foo(NODE) {}
     case "ArrowFunctionExpression":
     case "FunctionDeclaration":
     case "FunctionExpression":
+      if (parent.id === node) {
+        return false;
+      }
       for (const param of (parent.params: Array<any>)) {
         if (param === node) return false;
       }
-
-      return parent.id !== node;
+      return true;
```

Comment:
Can you add a `yes` comment which shows when this is true?

Reply:
I actually don't think there is a case.... This can all be simplified down to `return false`.

Reply:
Stand corrected, `ArrowFunctionExpression`s can have a referenced body identifier: `() => NODE`